### ISSUE TITLE
Function unable to take either Hash or named parameter variant

### ISSUE THREAD

**Issue by amirrajan** [mruby/mruby#6131]
The following works on mRuby `3.0.0`, `3.1.0`, CRuby `3.2.2`, and CRuby `3.4.0dev`, but raises an argument exception in mRuby `3.2.0`, and mRuby `master`:

mRuby `3.0.0`, CRuby `3.2.2`, CRuby `3.4.0dev`:
```ruby
def print_line line
  puts line
end

print_line({x: 0, y: 0, w: 100, h: 100}) # {:x=>0, :y=>0, :w=>100, :h=>100}
print_line(x: 0, y: 0, w: 100, h: 100) # {:x=>0, :y=>0, :w=>100, :h=>100}
```

mRuby `3.2.0`, `master`:
```ruby
def print_line line
  puts line
end

print_line({x: 0, y: 0, w: 100, h: 100}) # {:x=>0, :y=>0, :w=>100, :h=>100}
print_line(x: 0, y: 0, w: 100, h: 100) # wrong number of arguments (given 1, expected 1) (ArgumentError)
```

I performed a git bisect and it looks like the following commit caused the regression: [`5d1373335202a293d92eabf252223cab4559678f`](https://github.com/mruby/mruby/commit/5d1373335202a293d92eabf252223cab4559678f)

**Comment by dearblue** [comment]
I was able to reproduce it in mruby 3.2, but not in master (commit ebf66a2b801706eb02d801384c2d1fc84b579635).
I think this has been fixed since it was reported by #5944.

@amirrajan, could you check again to see if you have tried it with the latest master?

**Comment by amirrajan** [comment]
Confirmed. It is indeed fixed in `master` (my mistake)
